File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if(n==0):
            return 1
        len=0
        temp=n
        while(n>0):
            len+=1
            n=n>>1
        
        n=temp
        for i in range(len):
            p = 1<<i
            n=n^p

        return n
        

File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.java
class Solution {
    public int bitwiseComplement(int n) {
        if(n==0)
        return 1;
        int len = 0;
        int temp=n;
        while (n > 0) {
            len++;
            n >>= 1;
        }
        n=temp;
       // System.out.println(len);
        for(int i=0;i<len;i++)
        { 
            int pointer = 1<<i;
            n=n^pointer;
            //System.out.println(n);
        }
        return n;
    }
}